### PR TITLE
Establish git guidelines

### PR DESCRIPTION
**Pull request by tadaskay**

### CODE REVIEW

**Comment by soffokl** [comment]
Shouldn't it be a part of the contribution guide? 
http://docs.mysterium.network/en/latest/about/contributing/#style-guide

We are aimed to involve outside collaborators too, and it should not be limited to team members only.

**Comment by tadaskay** [comment]
This is aimed at a wider audience than `node` contribution guide. Even though it primarily targets team members, it is by no means limited to them only. We can later link to this from each project's contribution guide. Having this consolidated, it'll be easier to get onboard with company-wide and contribution practices.

File: README.md
Comment (soffokl) [inline README.md]: I'll point it here once again. 
Doing rebase for PR in GitHub does not keep history, and you have to review full PR from scratch, instead of only last commits. This is slowing down the PR review.

Comment (tadaskay) [reply README.md]: It does keep history, if you click on _force-pushed_ it will show the diff from the previous branch tip. Example for this PR: https://github.com/mysteriumnetwork/devbook/compare/cbda7bd5af50ca8833e02c08d81a433274848280..764fbaa4b372e766ccd92e0e73d9bbd933df6eb0

Comment (tadaskay) [reply README.md]: I think you are talking more about interactive rebase and editing existing commits. Rebase itself does nothing wrong.

Comment (soffokl) [reply README.md]: Yeah, I'm talking more about the review process, since commits were changed you don't know was it reviewed already or not.

File: README.md
Comment (donce) [inline README.md]: Please no branches with initials :D Also, `refactor/` is sometimes used as well :)

File: README.md
Comment (donce) [inline README.md]: 👍totally agree, I think this is where we need to improve :)

File: README.md
Comment (donce) [inline README.md]: Since we require all PR branches to be up to date with `master`, you often have to update your branch multiple times until it gets merged. After you already have approvals, rebasing it multiple times and getting approvals again is time-consuming for all and delays merging. So we usually are ok with creating merge commits for this case using web interface to avoid dismissing approvals.

File: README.md
Comment (donce) [inline README.md]: This is quite a lot of good rules, but we don't actually apply them all. I would suggest to start growing by small steps, not everything at once - i.e. we don't even always use first rule about imperative mood, so having half of these rules would be already a good start :)

What I mean, is we should have rules we can actually use, not rules we should use. It might be hard for all company to switch to best practices instantly.

File: README.md
Comment (donce) [inline README.md]: We can add an explanation about why one should use imperative mood - i.e. this has quite a good reference to git itself: https://stackoverflow.com/questions/3580013/should-i-use-past-or-present-tense-in-git-commit-messages/3580764#3580764

File: README.md
Comment (soffokl) [reply README.md]: > All practices in the devbook are meant to serve as guidelines, rather than rules.

File: README.md
Comment (vkuznecovas) [reply README.md]: We should then agree on SQUASHING to master...

Comment (soffokl) [reply README.md]: There is another option: don't worry about perfectly cleaned git history. :)

File: README.md
Comment (vkuznecovas) [inline README.md]: `tk/polish` is a bad example. Polish does not indicate anything that's meaningful and discernible, it's as good as `vk/breakfast`.

File: README.md
Comment (soffokl) [reply README.md]: Some PRs contains many commits and squashing it makes a single huge commit. Later it's starting to be harder to navigate through history with huge commits.

File: README.md
Comment (vkuznecovas) [reply README.md]: We'll need to consider options then - either remove the auto-stale'ing of reviews once it's been rebased(which seems bad) - or allow merging, but force squashing to master. If we want to move towards smaller PR's - it makes sense to squash to master.

File: README.md
Comment (vkuznecovas) [reply README.md]: not seeking for perfection, but seeing 7 merges in a row is a bit baffling.

Comment (donce) [reply README.md]: All approaches has pluses and minuses - personally I dislike squashing to master, because you get huge commits where all changes are merged, so it's hard to understand them and you can't revert single smaller commits.

I prefer rebase'ing, but merge commits are fine once you have a PR - it makes easier to review and saves time by keeping approvals.

File: README.md
Comment (tadaskay) [reply README.md]: Why not? Guys, let's be constructive and provide feedback instead of emotions or blank statements.

Reason for prefixes is to avoid clashes which may happen sometimes (from practice). I see nothing wrong here.

File: README.md
Comment (tadaskay) [reply README.md]: I dislike the auto-dismissal of reviews after force push. We don't have thirdparty contributions (as of now), and I don't see a problem if something is fixed up after I've given my thumbs up. Quite the opposite, I encourage fixing things without pinging back, unless explicitly needed.

File: README.md
Comment (tadaskay) [reply README.md]: If you're implementing breakfast, I see no problem with that. Small fixes should be encouraged, not discouraged, and if I can't afford half an hour deciding on the branch name, I'll just go with _polish_.

File: README.md
Comment (tadaskay) [reply README.md]: There's a link at the bottom of the section explaining everything in depth.

File: README.md
Comment (tadaskay) [reply README.md]: What Dmitrij said; plus these practices are well established, not from a blue sky. Link to the thoughbot guides is at the bottom of the page.

File: README.md
Comment (tadaskay) [reply README.md]: Squashing obfuscates history not less than tons of merge commits. Perhaps we can reconsider squashing _after_ we fully adopts small PRs?

File: git/README.md
Comment (tadovas) [inline git/README.md:44]: This! 👍

File: git/README.md
Comment (tadovas) [inline git/README.md:20]: 💯 karma points for short intro to rebase

**Review by tadovas — APPROVED** [review]
👍

**Review by iberflow — APPROVED** [review]

File: README.md
Comment (donce) [reply README.md]: Oh, that's why - I actually didn't have any painful experience with branch clashes. On the other hand - these initials feels like branch ownership, which might discourage collaboration, so I would suggest to use simply `feature/fix/refactor` or nothing in rare cases.

**Review by soffokl — APPROVED** [review]

**Review by vkuznecovas — APPROVED** [review]

**Pull request merged**
